File: swift_reference/Views/Onboarding/WelcomeScreen.swift
import SwiftUI

struct WelcomeScreen: View {
    let onContinue: () -> Void

    @State private var showGlow: Bool = false
    @State private var showTitle: Bool = false
    @State private var showTagline: Bool = false
    @State private var showVerse: Bool = false
    @State private var showButton: Bool = false
    @State private var breathe: Bool = false

    var body: some View {
        ZStack {
            TributeColor.charcoal.ignoresSafeArea()

            RadialGradient(
                colors: [
                    TributeColor.golden.opacity(showGlow ? 0.18 : 0),
                    TributeColor.golden.opacity(showGlow ? 0.06 : 0),
                    Color.clear
                ],
                center: .init(x: 0.5, y: 0.65),
                startRadius: 0,
                endRadius: 400
            )
            .ignoresSafeArea()
            .animation(.easeInOut(duration: 2.0), value: showGlow)

            EmberParticleView()
                .ignoresSafeArea()

            VStack(spacing: 0) {
                Spacer()

                VStack(spacing: 12) {
                    ZStack {
                        Circle()
                            .fill(
                                RadialGradient(
                                    colors: [
                                        Color.white.opacity(breathe ? 0.12 : 0.04),
                                        TributeColor.golden.opacity(breathe ? 0.25 : 0.08),
                                        Color.clear
                                    ],
                                    center: .center,
                                    startRadius: 0,
                                    endRadius: breathe ? 120 : 80
                                )
                            )
                            .frame(width: 280, height: 280)
                            .blur(radius: 40)

                        Text("TRIBUTE")
                            .font(.system(size: 52, weight: .bold, design: .serif))
                            .tracking(8)
                            .foregroundStyle(
                                LinearGradient(
                                    colors: [TributeColor.warmWhite, TributeColor.softGold],
                                    startPoint: .top,
                                    endPoint: .bottom
                                )
                            )
                            .shadow(color: TributeColor.golden.opacity(breathe ? 0.9 : 0.4), radius: breathe ? 35 : 18)
                            .shadow(color: TributeColor.golden.opacity(breathe ? 0.5 : 0.15), radius: breathe ? 70 : 35)
                    }
                    .opacity(showTitle ? 1 : 0)
                    .scaleEffect(showTitle ? 1 : 0.8)

                    Text("Track your habits. Give them to God.")
                        .font(.system(.body, design: .serif))
                        .foregroundStyle(TributeColor.softGold.opacity(0.85))
                        .opacity(showTagline ? 1 : 0)
                        .offset(y: showTagline ? 0 : 8)

                    VStack(spacing: 8) {
                        Text("\u{201C}Offer your bodies as a living sacrifice, holy and pleasing to God \u{2014} this is your true and proper worship.\u{201D}")
                            .font(.system(.subheadline, design: .serif))
                            .italic()
                            .foregroundStyle(TributeColor.softGold.opacity(0.55))
                            .multilineTextAlignment(.center)

                        Text("Romans 12:1")
                            .font(.caption)
                            .foregroundStyle(TributeColor.golden.opacity(0.45))
                    }
                    .padding(.top, 8)
                    .opacity(showVerse ? 1 : 0)
                    .offset(y: showVerse ? 0 : 10)
                }
                .padding(.horizontal, 40)

                Spacer()

                Button {
                    onContinue()
                } label: {
                    HStack(spacing: 8) {
                        Text("Let\u{2019}s begin")
                        Image(systemName: "arrow.right")
                            .font(.subheadline)
                    }
                    .tributeButton()
                }
                .padding(.horizontal, 40)
                .opacity(showButton ? 1 : 0)
                .offset(y: showButton ? 0 : 16)

                Spacer().frame(height: 56)
            }
        }
        .allowsHitTesting(true)
        .onAppear {
            withAnimation(.easeIn(duration: 1.5).delay(0.3)) {
                showGlow = true
            }

            withAnimation(.spring(duration: 0.9, bounce: 0.25).delay(0.6)) {
                showTitle = true
            }

            withAnimation(.easeOut(duration: 0.7).delay(1.0)) {
                showTagline = true
            }

            withAnimation(.easeOut(duration: 0.7).delay(1.4)) {
                showVerse = true
            }

            withAnimation(.easeOut(duration: 0.6).delay(1.8)) {
                showButton = true
            }

            withAnimation(.easeInOut(duration: 3.0).repeatForever(autoreverses: true)) {
                breathe = true
            }
        }
    }
}

File: lib/views/onboarding/welcome_screen.dart
import 'package:flutter/material.dart';
import '../../theme/app_theme.dart';

class WelcomeScreen extends StatefulWidget {
  final VoidCallback onNext;
  const WelcomeScreen({super.key, required this.onNext});

  @override
  State<WelcomeScreen> createState() => _WelcomeScreenState();
}

class _WelcomeScreenState extends State<WelcomeScreen>
    with SingleTickerProviderStateMixin {
  bool _showGlow = false;
  bool _showTitle = false;
  bool _showTagline = false;
  bool _showVerse = false;
  bool _showButton = false;
  late final AnimationController _breatheController;

  @override
  void initState() {
    super.initState();
    _breatheController = AnimationController(
      vsync: this,
      duration: const Duration(milliseconds: 3000),
    )..repeat(reverse: true);

    Future.delayed(const Duration(milliseconds: 300), () {
      if (mounted) setState(() => _showGlow = true);
    });
    Future.delayed(const Duration(milliseconds: 600), () {
      if (mounted) setState(() => _showTitle = true);
    });
    Future.delayed(const Duration(milliseconds: 1000), () {
      if (mounted) setState(() => _showTagline = true);
    });
    Future.delayed(const Duration(milliseconds: 1400), () {
      if (mounted) setState(() => _showVerse = true);
    });
    Future.delayed(const Duration(milliseconds: 1800), () {
      if (mounted) setState(() => _showButton = true);
    });
  }

  @override
  void dispose() {
    _breatheController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return AnimatedBuilder(
      animation: _breatheController,
      builder: (context, _) {
        final b = _breatheController.value;
        return Stack(
          fit: StackFit.expand,
          children: [
            AnimatedOpacity(
              opacity: _showGlow ? 1.0 : 0.0,
              duration: const Duration(milliseconds: 2000),
              child: Container(
                decoration: BoxDecoration(
                  gradient: RadialGradient(
                    colors: [
                      TributeColor.golden.withValues(alpha: 0.18),
                      TributeColor.golden.withValues(alpha: 0.06),
                      Colors.transparent,
                    ],
                    center: const Alignment(0, 0.3),
                    radius: 1.2,
                  ),
                ),
              ),
            ),
            SafeArea(
              child: Column(children: [
                const Spacer(),
                Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 40),
                  child: Column(children: [
                    AnimatedOpacity(
                      opacity: _showTitle ? 1.0 : 0.0,
                      duration: const Duration(milliseconds: 600),
                      child: AnimatedScale(
                        scale: _showTitle ? 1.0 : 0.8,
                        duration: const Duration(milliseconds: 900),
                        curve: Curves.elasticOut,
                        child: Stack(alignment: Alignment.center, children: [
                          Container(
                            width: 280,
                            height: 280,
                            decoration: BoxDecoration(
                              shape: BoxShape.circle,
                              gradient: RadialGradient(colors: [
                                Colors.white.withValues(alpha: 0.04 + b * 0.08),
                                TributeColor.golden.withValues(alpha: 0.08 + b * 0.17),
                                Colors.transparent,
                              ]),
                            ),
                          ),
                          ShaderMask(
                            shaderCallback: (bounds) => const LinearGradient(
                              colors: [TributeColor.warmWhite, TributeColor.softGold],
                              begin: Alignment.topCenter,
                              end: Alignment.bottomCenter,
                            ).createShader(bounds),
                            child: Text(
                              'TRIBUTE',
                              style: TextStyle(
                                fontSize: 52,
                                fontWeight: FontWeight.w700,
                                letterSpacing: 8,
                                color: Colors.white,
                                shadows: [
                                  Shadow(
                                    color: TributeColor.golden.withValues(alpha: 0.4 + b * 0.5),
                                    blurRadius: 18 + b * 17,
                                  ),
                                  Shadow(
                                    color: TributeColor.golden.withValues(alpha: 0.15 + b * 0.35),
                                    blurRadius: 35 + b * 35,
                                  ),
                                ],
                              ),
                            ),
                          ),
                        ]),
                      ),
                    ),
                    AnimatedOpacity(
                      opacity: _showTagline ? 1.0 : 0.0,
                      duration: const Duration(milliseconds: 700),
                      child: AnimatedSlide(
                        offset: _showTagline ? Offset.zero : const Offset(0, 0.3),
                        duration: const Duration(milliseconds: 700),
                        child: Text(
                          'Track your habits. Give them to God.',
                          textAlign: TextAlign.center,
                          style: TextStyle(
                            fontSize: 16,
                            color: TributeColor.softGold.withValues(alpha: 0.85),
                          ),
                        ),
                      ),
                    ),
                    const SizedBox(height: 20),
                    AnimatedOpacity(
                      opacity: _showVerse ? 1.0 : 0.0,
                      duration: const Duration(milliseconds: 700),
                      child: AnimatedSlide(
                        offset: _showVerse ? Offset.zero : const Offset(0, 0.4),
                        duration: const Duration(milliseconds: 700),
                        child: Column(children: [
                          Text(
                            '\u201COffer your bodies as a living sacrifice, holy and pleasing to God \u2014 this is your true and proper worship.\u201D',
                            textAlign: TextAlign.center,
                            style: TextStyle(
                              fontSize: 14,
                              fontStyle: FontStyle.italic,
                              color: TributeColor.softGold.withValues(alpha: 0.55),
                              height: 1.6,
                            ),
                          ),
                          const SizedBox(height: 6),
                          Text(
                            'Romans 12:1',
                            style: TextStyle(fontSize: 12, color: TributeColor.golden.withValues(alpha: 0.45)),
                          ),
                        ]),
                      ),
                    ),
                  ]),
                ),
                const Spacer(),
                AnimatedOpacity(
                  opacity: _showButton ? 1.0 : 0.0,
                  duration: const Duration(milliseconds: 600),
                  child: AnimatedSlide(
                    offset: _showButton ? Offset.zero : const Offset(0, 0.5),
                    duration: const Duration(milliseconds: 600),
                    child: Padding(
                      padding: const EdgeInsets.symmetric(horizontal: 40),
                      child: SizedBox(
                        width: double.infinity,
                        child: ElevatedButton.icon(
                          onPressed: widget.onNext,
                          icon: const Icon(Icons.arrow_forward_rounded, size: 16),
                          label: const Text("Let\u2019s begin",
                              style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16)),
                          style: ElevatedButton.styleFrom(
                            backgroundColor: TributeColor.golden,
                            foregroundColor: TributeColor.charcoal,
                            padding: const EdgeInsets.symmetric(vertical: 16),
                            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
                          ),
                        ),
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: 56),
              ]),
            ),
          ],
        );
      },
    );
  }
}
